torch/_inductor/fx_passes/pre_grad.py

```diff
@@ -128,6 +145,11 @@ def pre_grad_passes(gm: torch.fx.GraphModule, example_inputs=None):
                     gm,
                     f"[Pre grad(predispatch IR)]Apply split_cat, index: {ind}",
                 )
+            pass_execution_and_save(
+                remove_reshape_pass.apply,
+                gm,
+                "[Pre grad(predispatch IR)] Apply remove_reshape_pass",
+            )
```

Comment:
Noob q: if we've split the previously one `predispatch_pass` into three, should those three not be applied in a row (to preserve the BC)? I.e., should `remove_reshape_pass` not be applied directly after the other two, with nothing in-between?

Reply:
Good observation! This was actually intentional. Some passes relies on the graph being kwargs normalized while others currently don't (yet). Running a kwargs normalized graph on a pass that matches args renders that pass useless. That's why I move this pass towards the end (specifically after split_cat_aten) as it does kwargs normalization. In the meantime, we'll make all the passes to work with kwargs-normalized graph, because that's more reliable and easy to write passes.